modules/mongodb/src/main/java/org/testcontainers/containers/MongoDbContainer.java

```diff
@@ -0,0 +1,142 @@
+package org.testcontainers.containers;
+
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import lombok.NonNull;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.io.IOException;
+
+/**
+ * Constructs a single node MongoDB replica set for testing transactions.
+ * <p>To construct a multi-node MongoDB cluster, consider the <a href="https://github.com/silaev/mongodb-replica-set/">mongodb-replica-set project on GitHub</a>
+ * <p>Tested on a Mongo DB version 4.0.10+ (that is the default version if not specified).
+ *
+ * @author Konstantin Silaev on 9/30/2019
+ */
+@Slf4j
+public class MongoDbContainer extends GenericContainer<MongoDbContainer> {
+    private static final int CONTAINER_EXIT_CODE_OK = 0;
+    private static final int MONGODB_INTERNAL_PORT = 27017;
+    private static final int AWAIT_INIT_REPLICA_SET_ATTEMPTS = 60;
+    private static final String MONGODB_VERSION_DEFAULT = "4.0.10";
+    private static final String MONGODB_DATABASE_NAME_DEFAULT = "test";
+
+    public MongoDbContainer() {
+        super("mongo:" + MONGODB_VERSION_DEFAULT);
+        configureMongoDbContainer();
+
+    }
+
+    public MongoDbContainer(@NonNull final String dockerImageName) {
+        super(dockerImageName);
+        configureMongoDbContainer();
+    }
+
+    private void configureMongoDbContainer() {
+        withExposedPorts(MONGODB_INTERNAL_PORT);
+        withCommand("--replSet", "docker-rs");
+        waitingFor(
+            Wait.forLogMessage(".*waiting for connections on port.*", 1)
+        );
+    }
+
+    public String getReplicaSetUrl() {
+        if (!isRunning()) {
+            throw new IllegalStateException("MongoDbContainer should be started first");
+        }
+        return String.format(
+            "mongodb://%s:%d/%s",
+            getContainerIpAddress(),
+            getMappedPort(MONGODB_INTERNAL_PORT),
+            MONGODB_DATABASE_NAME_DEFAULT
+        );
+    }
+
+    @Override
+    public void start() {
+        super.start();
+        logReplicaSetStatus();
```

Comment:
this `execInContainer` will add 100-200ms to the startup time, especially noticeable while running with reusable containers mode. Given that it is for debugging only, I suggest we remove it.

Also, as a general note, consider not overriding `start` and using `containerIsStarting`/`containerIsStarted`

Reply:
done

Reply:
LGTM